#30 Add mobile-friendly elements to frontend NavBar

CodeDiversity opened the pull request:
## Description of Proposed Changes

Rssolves #30 

Adds mobile navbar for FE.

### Type of Change

_Please delete options that are not relevant._


- [x] New feature (non-breaking change which adds functionality)


## Testing

Added unit tests. 

## Checklists

### GitHub

- [x] Related issue appears at beginning of pull request title with pound sign, and title describes the changes being proposed.
- [x] Related issue is linked in pull request description using [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] The appropriate label has been chosen for this pull request.
- [x] The correct project has been selected for this pull request.
- [x] The milestone corresponding with the issue has been selected for this pull request.
- [x] All commits in this branch, including merge commits, begin with the issue number and a pound sign.

### Code Standards

- [x] Code follows the style guidelines of this project.
- [x] Code comments have been added, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] These changes generate no new warnings.

### Documentation and Testing

- [x] New and existing unit tests pass locally with these changes.
- [x] Corresponding changes to the documentation have been made.
- [x] Tests have been added, where appropriate; or, no tests are relevant for this pull request.
- [x] This pull request contains UI changes, and screenshots of those UI images appear below; or, this pull request contains no UI changes.

![image](https://github.com/Rainbow-Devs/health-tracker-frontend/assets/40902786/cdecd2d0-5406-4f49-82fb-5904910db9a5)

seidior committed:
#24 Improve eslint/prettier connection

Commit:
#30 adds mobile nav bar for the FE.

Commit:
#30 increase font size

seidior reviewed (changes requested):
Hi hi! I'm still looking at the code itself to review it, but I noticed that one of your commits doesn't _start_ with the issue number, even though the commit message does _contain_ it. Can you please modify it so it starts with the issue number?

Also, I have some questions:

- Is the menu built from scratch or is it based on the approach taken by one of the component libraries we've been using so far ([TailGrids](https://tailgrids.com/components), [Ariakit](https://ariakit.org)) as either inspiration or for importing the component functionality?

- You're importing three libraries as *runtime* dependencies (rather than dev dependencies) into the project for icon support. What went into your selection criteria for adding Font Awesome versus any other icon libraries, and are you using them in such a way that we can minimize the bundle size that is shipped to the client?

- Can you modify your contributors line to match the format seen in [the footer branch](https://github.com/Rainbow-Devs/health-tracker-frontend/blob/20-footer-ui/package.json)?

- It looks like you're mixing elements with Tailwind CSS styling, direct CSS styling, and styled-components styling. Can we consolidate these approaches to just Tailwind-based styling? (You can progressively get closer and closer by swapping out each CSS directive using `@apply`.)

- Finally, is there a reason you turned off "noImplicitAny" in the TypeScript settings rather than addressing the issue causing the implicit any warning/error?

Thanks!

Commit:
#30 making changes to use aria kit

seidior committed:
#30 Sync package.json from #32

seidior committed:
#30 Separate out hamburger SVG into component

seidior committed:
#30 Reorganize imports

seidior commented on components/organisms/NavBarWrapper.tsx:
```diff
@@ -0,0 +1,28 @@
+    };
+  }, []);
+  // if is mobile return hamburger menu else return navbar
+  return <>{isMobile ? <HamburgerMenu /> : <NavBar />}</>;
```
Two issues I see here:

1. This entirely removes the NavBar and its styling from the mobile design. I encourage you to come up with a solution here that only switches out the menu links for the menu button on small screen sizes. (Hint: display both the links and the menu button, and use Tailwind classes to enable one or the other based on screen size. Use CSS (via Tailwind class names) to handle the screen width, not JavaScript.)
2. We don't need the menu button yet since we don't have enough links to warrant it, so maybe let's get this to a shippable state and then comment it out until we need it?

seidior committed:
#30 Update latest from main

seidior committed:
#30 Merge latest from main

seidior committed:
#30 Remove redundant declaration in tsconfig

seidior committed:
#30 Prepare for Storybook merger

Moving only the new components to their own folders to better align with the changes made in structure in #27, as I expect it to merge before this PR does.

Everything else should move over automatically.

CodeDiversity commented:
Closing this for now, will re-do the story when having a mobile nav is warranted.
